server/item.go

```diff
@@ -12,7 +14,23 @@ type Item struct {
 	CreateAt int64  `json:"create_at"`
 }
 
-func itemsListToString(items []*Item) string {
+// ExtendedItem extends the information on Item to be used on the front-end
+type ExtendedItem struct {
+	Item
+	ForeignUser     string `json:"user"`
+	ForeignList     string `json:"list"`
+	ForeignPosition int    `json:"position"`
+}
```

Comment:
Nice use of embedding -- but consider just putting it all in the Item struct, since I assume we'll be using these Foreign keys all the time, right? If so, they're just a part of item.

Reply:
I am slightly against this because ForeignList and ForeignPosition may change during the time, and it is the "foreigner" the one changing those values. In order to avoid problems of conflicts in concurrency (two people modifying the same item at the same time) the items can only be added or removed. They never change during their lifetime.
I can add these fields to simplify the number of types in the code, but we would have to keep in mind that what is stored in the KV store is just a part, and the rest has to be recalculated.
One of the invariant parts at the moment would be the ForeignUser, but if we want in the future to create a "send" option for existing todos, or something similar, it will also change.

Reply:
Ok, sounds like a good reason to me. Let's keep it, so that you don't have to keep anything in mind. ("Keeping things in mind" is usually a bad thing.)